### PR TITLE
Include a stack trace along with failure to flush warning

### PR DESCRIPTION
**Pull request by idegtiarenko**
There are some number of such failures in the log, however they do not include the failure cause.
This change should propagate it.

### CODE REVIEW

**Review by DaveCTurner — APPROVED** [review]
LGTM

File: server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java [hunk 1]
```diff
@@ -2850,7 +2850,7 @@ public synchronized void afterMerge(OnGoingMerge merge) {
                     @Override
                     public void onFailure(Exception e) {
                         if (isClosed.get() == false) {
-                            logger.warn("failed to flush after merge has finished");
+                            logger.warn("failed to flush after merge has finished", e);
                         }
```
Comment (idegtiarenko) [inline server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java:2854]: I wonder if we should silently swallow the error if we are shutting down? 
May be at least log it at debug level?

Comment (DaveCTurner) [reply server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java:2854]: debug on shutdown would be fine with me too

**Review by DaveCTurner — APPROVED** [review]
LGTM

**Review by DaveCTurner — APPROVED** [review]
LGTM still

**Pull request merged**